cpnest/sampler.py

```diff
@@ -167,7 +177,27 @@ def estimate_nmcmc(self, safety=5, tau=None):
         self.Nmcmc = max(safety,int(self.Nmcmc_exact))
 
         return self.Nmcmc
+    
+    def estimate_nmcmc(self):
+        """
+        Estimate autocorrelation length of the chain
+        """
+        # first of all, build a numpy array out of
+        # the stored samples
+        self.ACL = []
+        samples = np.array(self.samples)
 
+        if self.store_chain is False:
+            self.ACL = [acl(samples[:,i]) for i in range(samples.shape[1])]
+        else:
+            self.ACL = [acl(samples[-self.maxmcmc:,i]) for i in range(samples.shape[1])]
+        self.Nmcmc = int(np.max(self.ACL))
+        if self.store_chain is False and len(self.samples) > self.maxmcmc:
```

Comment:
I don't understand the logic behind this. Can you add a comment?

Reply:
done.  the idea is that if you are storing the full mcmc chain, then you should use  only the last maxmcmc samples to compute the ACL and not the full story